CRM/Contact/Form/Task/EmailTrait.php

```diff
@@ -683,6 +684,27 @@ public static function saveTemplateFormRule(array $fields) {
     return empty($errors) ? TRUE : $errors;
   }
 
+  /**
+   * Prevent submission of deprecated tokens.
+   *
+   * @param array $fields
+   *
+   * @return bool|string[]
+   */
+  public static function deprecatedTokensFormRule(array $fields) {
+    $deprecatedTokens = [
+      '{case.status_id}' => '{case.status_id:label}',
+      '{case.case_type_id}' => '{case.case_type_id:label}',
+    ];
+    $tokenErrors = [];
+    foreach ($deprecatedTokens as $token => $replacement) {
+      if (strpos($fields['html_message'], $token) !== FALSE) {
+        $tokenErrors[] = ts('Token %1 is no longer supported - use %2 instead', ['{case.case_type_id}', '{case.case_type_id:label}']);
```

Comment:
Ok thanks. I didn't know ts() accepts 0-indexed placeholders - interesting.

The phrase "replaced token usage" seems weird to me. I'd be curious to see how that gets translated especially removed from context. Civi didn't replace anything. Is it even needed?

Reply:
@demeritcowboy any better wording idea?  - just "Unsupported tokens" ?

Reply:
Is it even needed?
If it feels to empty without it then sure "Unsupported tokens" sounds ok.

Reply:
@demeritcowboy ok - I took it out altogether - one less thing to translate